jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/parser/SettingsBodyParser.java

```diff
@@ -234,7 +239,7 @@ public void onConnectionFailure(int error, String reason)
             }
         });
         if (buffer.hasRemaining())
-            parser.parse(buffer);
+            parser.parse(buffer, 0, buffer.remaining());
```

Comment:
Why is the `buffer.remaining()`  value here equal to the bodyLength?   what if  you have a partial frame?

Reply:
Oh I see it is called only  with a synthetic SETTINGs body.    I see how it works, but it's just not obvious from the method/arg/field names.  It needs to be made much clearer that this can only be a full body parsed and there was no header.

Reply:
Added javadoc to the method.